aws/resource_aws_waf_geo_match_set_test.go

```diff
@@ -0,0 +1,342 @@
+					resource.TestCheckResourceAttr(
+						"aws_waf_geo_match_set.geo_match_set", "name", setName),
+					resource.TestCheckResourceAttr(
+						"aws_waf_geo_match_set.geo_match_set", "geo_match_constraints.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSWafGeoMatchSetDisappears(v *waf.GeoMatchSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).wafconn
+
+		wr := newWafRetryer(conn, "global")
+		_, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+			req := &waf.UpdateGeoMatchSetInput{
+				ChangeToken:   token,
+				GeoMatchSetId: v.GeoMatchSetId,
+			}
+
+			for _, geoMatchConstraint := range v.GeoMatchConstraints {
+				geoMatchConstraintUpdate := &waf.GeoMatchSetUpdate{
+					Action: aws.String("DELETE"),
+					GeoMatchConstraint: &waf.GeoMatchConstraint{
+						Type:  geoMatchConstraint.Type,
+						Value: geoMatchConstraint.Value,
+					},
+				}
+				req.Updates = append(req.Updates, geoMatchConstraintUpdate)
+			}
+			return conn.UpdateGeoMatchSet(req)
+		})
+		if err != nil {
+			return errwrap.Wrapf("[ERROR] Error updating GeoMatchSet: {{err}}", err)
+		}
+
+		_, err = wr.RetryWithToken(func(token *string) (interface{}, error) {
+			opts := &waf.DeleteGeoMatchSetInput{
+				ChangeToken:   token,
+				GeoMatchSetId: v.GeoMatchSetId,
+			}
+			return conn.DeleteGeoMatchSet(opts)
+		})
+		if err != nil {
+			return errwrap.Wrapf("[ERROR] Error deleting GeoMatchSet: {{err}}", err)
+		}
+		return nil
+	}
+}
+
+func testAccCheckAWSWafGeoMatchSetExists(n string, v *waf.GeoMatchSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No WAF GeoMatchSet ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafconn
+		resp, err := conn.GetGeoMatchSet(&waf.GetGeoMatchSetInput{
+			GeoMatchSetId: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if *resp.GeoMatchSet.GeoMatchSetId == rs.Primary.ID {
+			*v = *resp.GeoMatchSet
+			return nil
+		}
+
+		return fmt.Errorf("WAF GeoMatchSet (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccCheckAWSWafGeoMatchSetDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_waf_byte_match_set" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafconn
+		resp, err := conn.GetGeoMatchSet(
+			&waf.GetGeoMatchSetInput{
+				GeoMatchSetId: aws.String(rs.Primary.ID),
+			})
+
+		if err == nil {
+			if *resp.GeoMatchSet.GeoMatchSetId == rs.Primary.ID {
+				return fmt.Errorf("WAF GeoMatchSet %s still exists", rs.Primary.ID)
+			}
+		}
+
+		// Return nil if the GeoMatchSet is already destroyed
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "WAFNonexistentItemException" {
```

Comment:
Do you mind using a helper function here and simplify the code that way a bit? 😉 

```go
if isAWSErr(err, "WAFNonexistentItemException", "") {
```